### ISSUE TITLE
Fix support for linkattrs

### ISSUE THREAD

**Issue by lcawl** [elastic/docs#726]
Per https://github.com/elastic/docs/issues/505#issuecomment-459163236, I would like to use the linkattrs functionality described here: 

https://asciidoctor.org/docs/asciidoc-writers-guide/#target-window-and-role-attributes-for-links

When I try any of the examples there in the Glossary, for example:

> == Introduction
> 
> :linkattrs:
> 
> Chat with other Asciidoctor users on the http://discuss.asciidoctor.org/[*mailing list*^, role="green"].

 the attributes are not recognized. They're output as:

`<a class="ulink" href="http://discuss.asciidoctor.org/" target="_top"><span class="strong strong"><strong>mailing list</strong></span>^, role="green"</a>`

**Comment by nik9000** [comment]
I dug into this. It doesn't work for a whole stack of reasons:
1. Docbook 4.5 doesn't support it. Docbook 5 might, I'm not sure.
2. We force compat-mode on all of the documents. We probably shouldn't but we do. That seem like a thing for another time though. When we're in compat-mode we can't use linkatts.


We should fix compat-mode. Beyond that, though, we probably won't be able to use these until we take another look at docbook.

**Comment by nik9000** [comment]
I filed #728.

**Comment by lcawl** [comment]
You're right, it works when I disable compat-mode, enable linkattrs and use a default Asciidoctor -> HTML build. I have not tried with DocBook 5 yet.
